### PR TITLE
Random rotation port 

### PR DESCRIPTION
**Pull request by kamathis4**

### CODE REVIEW

File: keras_core/layers/preprocessing/random_rotation.py [hunk 1]
```diff
@@ -76,52 +68,161 @@ class RandomRotation(Layer):
...
+                image_height = inputs.shape[0]
+                image_width = inputs.shape[1]
+            else:
+                image_height = inputs.shape[1]
+                image_width = inputs.shape[2]
+
+        lower = self._factor[0] * 2.0 * backend.convert_to_tensor(np.pi)
+        upper = self._factor[1] * 2.0 * backend.convert_to_tensor(np.pi)
+
+        angle = backend.random.uniform(shape=(batch_size,), minval=lower, maxval=upper, dtype=np.float32)
```
Comment (fchollet) [inline keras_core/layers/preprocessing/random_rotation.py:174]: You need to pass a seed generator, otherwise you get the same value every time.

Use `seed_generator = self._get_seed_generator(self.backend._backend)` then `seed=seed_generator`

File: keras_core/layers/preprocessing/random_rotation.py [hunk 2]
```diff
@@ -76,52 +68,161 @@ class RandomRotation(Layer):
...
+
+        y_offset = (
+            (image_height - 1)
+            - (
+                sin_theta * (image_width - 1)
+                + cos_theta * (image_height - 1)
+            )
+        ) / 2.0
+
+        return backend.numpy.concatenate([
```
Comment (fchollet) [inline keras_core/layers/preprocessing/random_rotation.py:195]: All calls to the backend should use `self.backend` in order to ensure we can set the backend dynamically in the TFData use case

**Comment by kamathis4** [comment]
For some reason I'm getting an extra '__annotations__' while running layer test, am I missing something?

File: keras_core/layers/preprocessing/random_rotation.py [hunk 3]
```diff
@@ -76,52 +68,186 @@ class RandomRotation(Layer):
...
+        )
+        if input_shape == 3:
+            outputs = self.backend.numpy.squeeze(outputs, axis=0)
         return outputs
 
+    def call(self, inputs, training=True):
+        inputs = self.backend.cast(inputs, self.compute_dtype)
+        if training:
+            rotation_matrix = self._get_rotation_matrix(inputs)
+            transformed_image = affine_transform(
```
Comment (james77777778) [inline keras_core/layers/preprocessing/random_rotation.py:227]: The failed tests should be fixed by replacing `affine_transform` from keras_core.ops.image with `self.backend.image.affine_transform`

**Review by james77777778 — COMMENTED** [review]
I just came across this PR :)

File: keras_core/layers/preprocessing/random_rotation.py (hunk 3 above)
Comment (kamathis4) [reply keras_core/layers/preprocessing/random_rotation.py:227]: Hey @james77777778, thanks for the help!!

Comment (kamathis4) [reply keras_core/layers/preprocessing/random_rotation.py]: I'm getting a really small difference in the element which is supposed to be 0, in the torch implementation, about 2.62260437e-06. Is it wise to increase rtol?
Also this is after importing the torch affine transform function updation.
What do you suggest I do?
cc: @fchollet

Comment (fchollet) [reply keras_core/layers/preprocessing/random_rotation.py]: > Is it wise to increase rtol?

Yes, any numerical difference that is close to the float32 precision is fine to write off. You can use a 1e-5 rtol.

**Review by fchollet — COMMENTED** [review]
Thanks for the update. The code is looking good!

**Review by fchollet — APPROVED** [review]
Thank you for the great contribution! LGTM
